stress_test/nb_generator.py

```diff
@@ -0,0 +1,254 @@
+
+        self.cpus = test_config['nb_generator_cpu_shares']
+        # self.host_spec = test_config['nb_generator_node_spec']
+        # self.host_ip = test_config['nb_generator_host_ip']
+
+        self.ip = test_config['nb_generator_ode_ip']
+        self.ssh_port = test_config['nb_generator_node_ssh_port']
+        self.ssh_user = test_config['nb_generator_node_username']
+        self.ssh_pass = test_config['nb_generator_node_password']
+
+        self.run_hnd = (self.base_dir +
+                        test_config['nb_generator_run_handler'])
+        self.status = 'UNKNOWN'
+        self._ssh_conn = None
+
+        self.flow_delete_flag = test_config['flow_delete_flag']
+        self.flows_per_request = test_config['flows_per_request']
+        self.log_level
+
+        # The parameters initialized as None are dimensions of the test.
+        # These values are passed outside, from the test in the main for loop.
+        # ---------------------------------------------------------------------
+        self.flow_workers = None
+        self.total_flows = None
+        self.flow_operations_delay_ms = None
+        # ---------------------------------------------------------------------
+
+    def init_ssh(self):
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name, self.ip))
+
+        self._ssh_conn = \
+            util.netutil.ssh_connect_or_return2(self.ip,
+                                                int(self.ssh_port),
+                                                self.ssh_user,
+                                                self.ssh_pass,
+                                                10)
+
+    def run(self):
+        """ Wrapper to the NB-Generator run handler
+        :returns: Returns the combined stdout - stderr of the executed command
+        :rtype: str
+        """
+        logging.info("[NB_generator] Run handler")
+        cmd = ('cd {0}; taskset -c {1} python3.4 {2} {3} {4} {5} {6} {7} {8} {9} {10} {11} {12}'.
+               format(self.base_dir,
+                      self.cpus,
+                      self.run_hnd,
+                      self.controller.ip,
+                      self.controller.restconf_port,
+                      self.total_flows,
+                      self.flow_workers,
+                      self.flow_operations_delay_ms,
+                      self.flow_delete_flag,
+                      self.controller.restconf_user,
+                      self.controller.restconf_pass,
+                      self.flows_per_request,
+                      self.log_level))
+        logging.debug('Generator handler command:{0}.'.format(cmd))
+
+        exit_status, output = \
+            util.netutil.ssh_run_command(self._ssh_conn,
+                                         cmd,
+                                         '[NB_generator_run_handler]')
+        if exit_status == 0:
+            self.status = 'RUNNING'

```

Comment:
is the run handler of the NB generator blocking or non-blocking? 

i.e., when `ssh_run_handler` returns, has the NB gen finished sending flows, or has it just started running? 

if we are in the 1st case, we have to update appropriately the statuses and log messages (e.g. to `FINISHED` and `FAILED`)